qtrio/_qt.py

```diff
@@ -25,20 +26,34 @@ class Signal:
     _attribute_name: typing.ClassVar[str] = ""
 
     def __init__(self, *args: object, **kwargs: object) -> None:
+        from qts import QtCore
+
         class _SignalQObject(QtCore.QObject):
             signal = QtCore.Signal(*args, **kwargs)
 
         self.object_cls = _SignalQObject
 
-    def __get__(self, instance: object, owner: object) -> qtrio._util.SignalInstance:
+    @typing.overload
+    def __get__(self, instance: None, owner: object) -> typing.Union["Signal"]:
+        ...
+
+    @typing.overload
+    def __get__(
+        self, instance: object, owner: object
+    ) -> typing.Union["QtCore.SignalInstance"]:
+        ...
```

Comment:
is something supposed to be here?

Reply:
No, these are just type hint overloads that let us express in what cases you will get which return type.  Without them you end up having to always handle both types when exercising the `.__get__()` even though the single return type is defined by the usage.

Reply:
I suppose I could share the [mypy-play.net](https://mypy-play.net/?mypy=latest&python=3.10&gist=a8db09996a187e7efaa671692d36fd57) example I made up to pre-test this setup.  Also [the documentation for the `.__get__()`](https://docs.python.org/3.9/reference/datamodel.html#object.__get__) bit of the descriptor protocol.  Also, for some reason I haven't read up on it is common to use [`...` (ellipsis)](https://docs.python.org/3.9/library/constants.html#Ellipsis) in type hinting situations like this.  You can see it throughout the examples in the [typing](https://docs.python.org/3.9/library/typing.html) module.

```python
import typing


class D:
    @typing.overload
    def __get__(self, instance: None, owner: object) -> "D":
        ...

    @typing.overload
    def __get__(self, instance: object, owner: object) -> int:
        ...

    def __get__(self, instance: object, owner: object) -> typing.Union["D", int]:
        ...


class C:
    d = D()

reveal_type(C.d)
reveal_type(C().d)
```

```
main.py:20: note: Revealed type is "main.D"
main.py:21: note: Revealed type is "builtins.int"
```